fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexTest.java

```diff
@@ -612,6 +612,309 @@ void partitionedJoinedIndexTest() {
+            }
+
+            // 30 records in partition 1
+            for (int i = 0; i < 30; i++) {
+                recordStore.saveRecord(createComplexDocument(2222L + i, ENGINEER_JOKE, 1, Instant.now().minus(20, ChronoUnit.DAYS).toEpochMilli()));
+            }
+
+            LuceneContinuationProto.LuceneIndexContinuation continuation = LuceneContinuationProto.LuceneIndexContinuation.newBuilder()
+                    .setPartitionId(1)
+                    .setPartitionTimestamp(timestamp29DaysAgo)
+                    .setDoc(16)
+                    .setScore(0.012435136F)
+                    .setShard(0)
+                    .build();
+
+            assertEquals(43, recordStore.scanIndex(COMPLEX_PARTITIONED, groupedTextSearch(COMPLEX_PARTITIONED, "text:propose", 1), continuation.toByteArray(), ExecuteProperties.newBuilder().setReturnedRowLimit(50).build().asScanProperties(false))
+                    .getCount().join());
+
+            continuation = LuceneContinuationProto.LuceneIndexContinuation.newBuilder()
+                    .setPartitionId(1)
+                    .setPartitionTimestamp(timestamp60DaysAgo)
+                    .setDoc(16)
+                    .setScore(0.012435136F)
+                    .setShard(0)
+                    .build();
+
+            final Subspace partition0Subspace = recordStore.indexSubspace(COMPLEX_PARTITIONED).subspace(Tuple.from(1, LucenePartitioner.PARTITION_DATA_SUBSPACE).add(0));
+            final Subspace partition1Subspace = recordStore.indexSubspace(COMPLEX_PARTITIONED).subspace(Tuple.from(1, LucenePartitioner.PARTITION_DATA_SUBSPACE).add(1));
+
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition0Subspace, context, "_0.cfs");
+            validateSegmentAndIndexIntegrity(COMPLEX_PARTITIONED, partition1Subspace, context, "_0.cfs");
+        }
+    }
+
+    /**
+     * test cross partition limit query with multiple scans.
+     */
+    @Test
+    void testPartitionedLimitNeedsMultipleScans() {
+        try (FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, COMPLEX_DOC, COMPLEX_PARTITIONED);
+
+            long timestamp60DaysAgo = Instant.now().minus(60, ChronoUnit.DAYS).toEpochMilli();
+            long timestamp29DaysAgo = Instant.now().minus(29, ChronoUnit.DAYS).toEpochMilli();
+
+            // partition 0 covers 30 days beginning 60 days ago
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 0,
+                    timestamp60DaysAgo,
+                    Instant.now().minus(30, ChronoUnit.DAYS).toEpochMilli());
+
+            // partition 1 covers the past 29 days till yesterday
+            createPartitionMetadata(COMPLEX_PARTITIONED, Tuple.from(1), 1,
+                    timestamp29DaysAgo,
+                    Instant.now().minus(1, ChronoUnit.DAYS).toEpochMilli());
+
+            // 30 records in partition 0
```

Comment:
30 is a copy-paste typo. 300 is the intent, since the original non-partitioned test case used a value > scan page size. The other option is to the change the value of the page size property for the test and create fewer records.